src/memory-analyzer/gdb_api.cpp

```diff
@@ -107,15 +107,28 @@ void gdb_apit::create_gdb_process()
     dup2(pipe_output[1], STDOUT_FILENO);
     dup2(pipe_output[1], STDERR_FILENO);
 
-    dprintf(pipe_output[1], "binary name: %s\n", binary);
+    dprintf(pipe_output[1], "binary name: %s\n", args.front().c_str());
 
-    char *const arg[] = {const_cast<char *>("gdb"),
-                         const_cast<char *>("--interpreter=mi"),
-                         const_cast<char *>(binary),
-                         NULL};
+    std::vector<std::string> exec_cmd;
+    exec_cmd.reserve(args.size() + 3);
+    exec_cmd.push_back("gdb");
+    exec_cmd.push_back("--interpreter=mi");
+    exec_cmd.push_back("--args");
+    exec_cmd.insert(exec_cmd.end(), args.begin(), args.end());
+
+    char **exec_cmd_ptr = (char **)malloc(
```

Comment:
Now that we build these using `malloc`, should we free the memory at some point?

Reply:
Yes I think we could free it in the parent process.

Reply:
Ah I just realised we're allocating memory after the `fork()` call in the child process only. And that memory is freed when the process image is replaced via `execvp()`.